packages/explorer/src/app/(explorer)/[chainName]/worlds/WorldsForm.tsx

```diff
@@ -25,6 +27,9 @@ const formSchema = z.object({
 });
 
 export function WorldsForm({ worlds }: { worlds: Address[] }) {
+  // Initialize the observer store to start fetching transactions
+  useStore(store);
```

Comment:
does this cause a rerender of this component every time the store changes?

Reply:
I wonder if the zustand suggested approach of using a provider in next.js is for this reason 🙈 I assumed it was an SSR thing but maybe its a broad solution to this particular issue

Reply:
yeah, unfortunately causes re-renders, it's definitely a code smell to say the least. Using the provider does feel cleaner despite the all the boilerplate. I think the Provider solution is specifically for SSR because I'm not sure how common of a scenario it is to initialize a store in such way. Usually, you'd use one of store's getters/setters, and initialize the store implicitly when it's needed.

Given that the current problem essentially is that the broadcast channel listener does get started, I'm thinking what about moving the broadcast channel event listener into a hook, and initialize it somewhere higher-up the tree that way?

Reply:
do you know if the provider would solve for this?

if so I feel like maybe the provider approach is ~fine and solves for multiple problems (SSR, global mounting/initializing the store and observer)

but curious if we can generalize the store provider boilerplate to avoid repeating the pattern for every store

Reply:
I'd need to set it up to double-check but I'd think so. Given though that the `store` solution works, even if somewhat hacky, I'm included to stick with that